lib/Net/IPManager.pm

```diff
@@ -131,36 +143,51 @@ sub _is_in ($self, $ip, $range) {
 Checks whether C<$ips_string> looks like a valid ip in CIDR notation
 
 =cut
-sub looks_like_ip ($self, $ips_string) {
+sub looks_like_ip($self, $ips_string) {
     my @ips = map {_trm($_)} split /\,/, $ips_string;
     for my $ip (@ips) {
         my $may_ip = Net::IP->new($ip) or undef;
     }
     return 1;
 }
 
+=head3 get_next_ip($interface_range [, $offset = 1])
+
+Takes a subnet (larger than /32 or /128) and returns the IP with $offset away from the network address.
+
+=cut
+sub get_next_ip($interface_range, $offset = 1) {
+    my $prefix = $interface_range->version() == 4 ? '/32' : '/128';
+    my $next_ip = $interface_range->ip_add_num($offset);
+    # This is unfortunately necessary since ip_add_num() keeps the subnet size
+    return defined $next_ip ? Net::IP->new($next_ip->ip() . $prefix) : undef;
+}
+
 =head3 suggest_ip($interface)
 
 Returns a comma separated string of the next free ip(s) for C<$interface>. If no ips are left, an empty string is returned
 
 =cut
-sub suggest_ip ($self, $interface) {
+sub suggest_ip($self, $interface) {
     my @suggested_ips;
-    for my $interface_range (@{$self->interface_ranges->{$interface}}) {
+    for my $interface_range (@{$self->{interface_ranges}->{$interface}}) {
```

Comment:
```suggestion
    for my $interface_range ($self->{interface_ranges}{$interface}->@*) {
```
looks nicer finde ich

Reply:
I'm leaving this as-is for readability  and consistency